### PR TITLE
Update tests and fix solution for P20

### PR DESCRIPTION
**Pull request by amanketebo**
* Add an assert to the randomInt function to make sure division by 0 doesn't happen

### CODE REVIEW

**Comment by amanketebo** [comment]
Hey @eneko, 
In regard to the Travis CI build failing, you'll notice that the osx build passes but the linux build fails. I tracked down the problem and found out its related to this random number function:

```swift
public func randomInt(under limit: Int) -> Int {
    #if os(Linux)
        assert(limit != 0)
        srandom(UInt32(time(nil)))
        return random() % limit
    #else
        return Int(arc4random_uniform(UInt32(limit)))
    #endif
}
```

I believe the linux build is failing in part because of the updates I made to the tests since they were commented out before. The build fails while testing your solution to problem 23. More specifically, the test involving a list length of 1. Your solution on a linux environment ends up doing a division by zero. I just wanted to give you a heads up on why it was failing on one environment and not the other. Thanks!

**Comment by eneko** [comment]
Thank you, will take a look!

**Comment by eneko** [comment]
Hello @amanketebo, 

I've looked into the issue and it seems like it was an oversight on my end when I wrote the `randomInt()` method. Per definition, when limit is less than 2 (0 or 1, this is), the result should always be zero.

I've fixed this in https://github.com/eneko/Ninety-Nine-Swift-Solutions/pull/32. Please update your PR with those changes, tests should pass afterwards.

Thanks!
Eneko

**Comment by amanketebo** [comment]
Nice work 👍🏾It looks you also updated the branch, thanks!

File: Sources/NinetyNineSwiftProblems/List.swift [hunk 1]
```diff
@@ -30,6 +38,19 @@ extension List: CustomStringConvertible {
...
+        while let currentValue = current?.value {
+            if currentValue == value {
+                return true
+            }
+            current = current?.nextItem
+        }
+        return false
+    }
+}
+
```
Comment (eneko) [inline Sources/NinetyNineSwiftProblems/List.swift]: I'm thinking we shouldn't add these helper methods to the default `List` class. `forEach` is a nice one, and so it is `contains` (this one I believe is another programming challenge). I'm thinking also on removing the `description` helper.

These helper methods can be moved so they are available only on the tests, but not on the solutions. The goal is to let developers solve challenges their own way, without providing any helper methods other than the initializers.

For now, please move these two helper methods to the test file where they are being used (`p24_tests.swift`).

Comment (amanketebo) [reply Sources/NinetyNineSwiftProblems/List.swift]: Whoops, I didn't think about that. Yeah I totally agree we should take those functions out of the default `List` class. I'll go ahead and do that right now.

**Comment by eneko** [comment]
Nice, thanks for the updates!

**Pull request merged**